src/sdp_proxy/advertising_proxy.cpp

```diff
@@ -302,15 +301,17 @@ otbrError AdvertisingProxy::PublishHostAndItsServices(const otSrpServerHost *aHo
 
     if (!hostDeleted)
     {
+        std::vector<uint8_t> firstHostAddress{hostAddress[0].mFields.m8,
+                                              hostAddress[0].mFields.m8 + sizeof(hostAddress[0])};
```

Comment:
For simplicity, you can use `std::end(hostAddress[0].mFields.m8)` for the end iterator. I'm not sure if `std::end` is considered allowed in the codebase though.

Reply:
updated!